File: src/main/java/vjson/listener/AbstractParserListener.java
/*
 * The MIT License
 *
 * Copyright 2019 wkgcass (https://github.com/wkgcass)
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

package vjson.listener;

import vjson.JSON;
import vjson.ParserListener;
import vjson.parser.*;

import java.util.List;
import java.util.Map;

public abstract class AbstractParserListener implements ParserListener {
    @Override
    public void onObjectBegin(ObjectParser object) {

    }

    @Override
    public void onObjectKey(ObjectParser object, String key) {

    }

    @Override
    public void onObjectValue(ObjectParser object, String key, JSON.Instance value) {

    }

    @Override
    public void onObjectValueJavaObject(ObjectParser object, String key, Object value) {

    }

    @Override
    public void onObjectEnd(ObjectParser object) {

    }

    @Override
    public void onObject(JSON.Object object) {

    }

    @Override
    public void onObject(Map<String, Object> object) {

    }

    @Override
    public void onArrayBegin(ArrayParser array) {

    }

    @Override
    public void onArrayValue(ArrayParser array, JSON.Instance value) {

    }

    @Override
    public void onArrayValueJavaObject(ArrayParser array, Object value) {

    }

    @Override
    public void onArrayEnd(ArrayParser array) {

    }

    @Override
    public void onArray(JSON.Array array) {

    }

    @Override
    public void onArray(List<Object> array) {

    }

    @Override
    public void onBoolBegin(BoolParser bool) {

    }

    @Override
    public void onBoolEnd(BoolParser bool) {

    }

    @Override
    public void onBool(JSON.Bool bool) {

    }

    @Override
    public void onBool(Boolean bool) {

    }

    @Override
    public void onNullBegin(NullParser n) {

    }

    @Override
    public void onNullEnd(NullParser n) {

    }

    @Override
    public void onNull(JSON.Null n) {

    }

    @Override
    public void onNull(Void n) {

    }

    @Override
    public void onNumberBegin(NumberParser number) {

    }

    @Override
    public void onNumberFractionBegin(NumberParser number, long integer) {

    }

    @Override
    public void onNumberExponentBegin(NumberParser number, double base) {

    }

    @Override
    public void onNumberEnd(NumberParser number) {

    }

    @Override
    public void onNumber(JSON.Number number) {

    }

    @Override
    public void onNumber(Number number) {

    }

    @Override
    public void onStringBegin(StringParser string) {

    }

    @Override
    public void onStringChar(StringParser string, char c) {

    }

    @Override
    public void onStringEnd(StringParser string) {

    }

    @Override
    public void onString(JSON.String string) {

    }

    @Override
    public void onString(String string) {

    }

    @Override
    public void onError(String err) {

    }
}

File: base/src/main/kotlin/vjson/listener/AbstractParserListener.kt
/*
 * The MIT License
 *
 * Copyright 2021 wkgcass (https://github.com/wkgcass)
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */
package vjson.listener

import vjson.JSON
import vjson.ParserListener
import vjson.parser.*

abstract class AbstractParserListener : ParserListener {
  override fun onObjectBegin(obj: ObjectParser) {}
  override fun onObjectKey(obj: ObjectParser, key: String) {}
  override fun onObjectValue(obj: ObjectParser, key: String, value: JSON.Instance<*>) {}
  override fun onObjectValueJavaObject(obj: ObjectParser, key: String, value: Any?) {}
  override fun onObjectEnd(obj: ObjectParser) {}
  override fun onObject(obj: JSON.Object) {}
  override fun onObject(obj: Map<String, *>?) {}
  override fun onArrayBegin(array: ArrayParser) {}
  override fun onArrayValue(array: ArrayParser, value: JSON.Instance<*>) {}
  override fun onArrayValueJavaObject(array: ArrayParser, value: Any?) {}
  override fun onArrayEnd(array: ArrayParser) {}
  override fun onArray(array: JSON.Array) {}
  override fun onArray(array: List<*>?) {}
  override fun onBoolBegin(bool: BoolParser) {}
  override fun onBoolEnd(bool: BoolParser) {}
  override fun onBool(bool: JSON.Bool) {}
  override fun onBool(bool: Boolean) {}
  override fun onNullBegin(n: NullParser) {}
  override fun onNullEnd(n: NullParser) {}
  override fun onNull(n: JSON.Null) {}
  override fun onNull(n: Unit?) {}
  override fun onNumberBegin(number: NumberParser) {}
  override fun onNumberFractionBegin(number: NumberParser, integer: Long) {}
  override fun onNumberExponentBegin(number: NumberParser, base: Double) {}
  override fun onNumberEnd(number: NumberParser) {}
  override fun onNumber(number: JSON.Number<*>) {}
  override fun onNumber(number: Number) {}
  override fun onStringBegin(string: StringParser) {}
  override fun onStringChar(string: StringParser, c: Char) {}
  override fun onStringEnd(string: StringParser) {}
  override fun onString(string: JSON.String) {}
  override fun onString(string: String) {}
  override fun onError(err: String) {}
}
